src/contexts/UserContext.js

```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext } from "react";
+import { useUserData } from "../custom-hooks/useUserData";
+
+const UserContext = createContext();
+
+const useUserContext = () => useContext(UserContext);
+
+const UserProvider = (props) => {
+  const userObj = useUserData();
+  return (
+    <UserContext.Provider value={{ ...userObj }}>
```

Comment:
why spread here ?
are you passing object properties instead of object ?

suggestion pass the whole object and destructure where ever needed